chapters/ch05.asciidoc

```diff
@@ -207,7 +211,14 @@ Let's move onto a more interesting topic: asynchronous code flows.
 
 ==== 5.1.4 Navigating Callbacks, Promises, and Asynchronous Functions
 
-JavaScript now offers several options when it comes to describing asynchronous algorithms: the plain callback pattern, promises, async functions, async iterators, async generators, plus any patterns offered by libraries consumed in our applications. Each solution comes with their own set of strengths and weaknesses: callbacks are typically a solid choice, but we often need to get libraries involved when we want to execute our work concurrently; Promises might be hard to understand at first, but they offer a few utilities like `Promise#all` for concurrent work, yet they might be hard to debug under some circumstances; Async functions require a bit of understanding on top of being comfortable with promises, but they're easier to debug and often result in simpler code, plus they can be interspersed with synchronous functions rather easily as well; Iterators and generators are powerful tools, but there aren't many practical use cases for them, so we must consider whether we're using them because they fit our needs or just because we can.
+JavaScript now offers several options when it comes to describing asynchronous algorithms: the plain callback pattern, promises, async functions, async iterators, async generators, plus any patterns offered by libraries consumed in our applications. Each solution comes with its own set of strengths and weaknesses:
+
+1. Callbacks are typically a solid choice, but we often need to get libraries involved when we want to execute our work concurrently;
+1. Promises might be hard to understand at first, but they offer a few utilities like `Promise#all` for concurrent work, yet they might be hard to debug under some circumstances;
+1. Async functions require a bit of understanding on top of being comfortable with promises, but they're easier to debug and often result in simpler code, plus they can be interspersed with synchronous functions rather easily as well;
+1. Iterators and generators are powerful tools, but there aren't many practical use cases for them, so we must consider whether we're using them because they fit our needs or just because we can.
+
+> there aren’t many practical use cases for them? [citation needed] I for one use iteration all the time.
```

Comment:
True, it's mostly that often people use these new shiny things just because they can, changed to `all that many`